system/include/emscripten/val.h

```diff
@@ -570,7 +570,7 @@ class val {
   }
 
 // If code is not being compiled with GNU extensions enabled, typeof() is not a reserved keyword, so support that as a member function.
-#if __STRICT_ANSI__
+#if __is_identifier(typeof)
```

Comment:
Comment needs updating as well?

Reply:
I think comment is still correct, it's just that the check wasn't extensive enough.

Reply:
Shouldn't this be inverted?   i.e. should the member declared when `typeof` is *not* a reserved keyword?

Reply:
That's what it does: `__is_identifier` checks if something can be used as a valid identifier, that is, that it's not a reserved word.

Reply:
Ah I see..   Shame we can't use `if !__is_keyword(..)` which would be more descriptive I think.

Reply:
Yeah or the name could be `__is_valid_identifier`. Still nice that they have this macro at all.